backend/src/controllers/reviewController.ts

```diff
@@ -23,6 +25,37 @@ export const addReview = async (req: Request, res: Response) => {
   }
 }
 
+export const deleteReview = async (req: Request, res: Response) => {
+  const jwt = req.cookies.authToken;
+  if (!jwt) {
+    res.status(400).json({ message: 'Missing JWT cookie' });
+  }
+  let id: number;
+  try {
+    const payload = auth.verifyAndReadJWT(jwt);
+    id = payload.id;
+
+    const review = req.body;
+    if (!review || !review.id || !review.account_id) {
+      res.status(400).json({ message: 'Invalid review data' });
+    }
+
+    try {
+      const account = await accountModel.getAccount(id);
+      if (account?.id !== review.account_id) {
+        res.status(403).json({ message: 'You are not authorized to delete this review' });
+      }
```

Comment:
if you think about what you're doing rn, you're fetching the account by the id to get the id... you already have it, no need for the query.